packages/venia-concept/src/ServiceWorker/Utilities/__tests__/htmlHandler.spec.js

```diff
@@ -0,0 +1,141 @@
+jest.mock('../messageHandler');
+
+import { HTML_UPDATE_AVAILABLE } from '@magento/venia-ui/lib/constants/swMessageTypes';
+
+import { cacheHTMLPlugin } from '../htmlHandler';
+import { sendMessageToWindow } from '../messageHandler';
+
+const matchFn = jest.fn();
+
+beforeAll(() => {
+    global.caches = {
+        match: matchFn
+    };
+    global.Request = function(url, { method, headers } = {}) {
+        this.url = url;
+        this.method = method;
+        this.headers = headers;
+    };
+    global.Response = function() {};
+});
+
+beforeEach(() => {
+    sendMessageToWindow.mockClear();
+});
+
+test('cacheHTMLPlugin should have cacheKeyWillBeUsed, requestWillFetch, fetchDidSucceed async functions implemented', () => {
+    matchFn.mockReturnValue(Promise.resolve(null));
+
+    expect(cacheHTMLPlugin).toHaveProperty('cacheKeyWillBeUsed');
+    expect(cacheHTMLPlugin).toHaveProperty('requestWillFetch');
+    expect(cacheHTMLPlugin).toHaveProperty('fetchDidSucceed');
```

Comment:
Just to save yourself some time in the future, these aren't _strictly_ necessary because your next set of assertions will fail if these properties aren't present.

But it reads better like this, so it's likely a personal preference thing.

Reply:
I needed tests that will fail if in the future someone accidentally changes them without them pondering about what happened. Also, I feel it should be a norm usually to have tests around testing the shape of an entity like snapshots do for react components.

Reply:
Yeah I'm just saying the assertions immediately below this serve the same purpose.